### ISSUE TITLE
New cop: `Lint/InfiniteRecursion`

### ISSUE THREAD

**Issue by fatkodima** [rubocop-hq/rubocop#8553]
This cop should detect possible infinite recursions within methods.

```ruby
# bad
def factorial(n)
  n * factorial(n - 1)
end

# good
def factorial(n)
  if n <= 1
    1
  else
    n * factorial(n - 1)
  end
end
```

**Comment by marcandre** [comment]
The issue I see is that the simple cases will simply not be found in really code as they won't ever work.

**Comment by fatkodima** [comment]
In well tested public code this for sure *should not* find anything 😄 
In bad proprietary code - maybe.

I have made a POC of this several weeks ago, do not remember what exactly went wrong. Maybe worth a second try.

**Comment by fatkodima** [comment]
Decided, so this is not worth the effort.
